lib/async/http/faraday/adapter.rb

```diff
@@ -71,6 +78,18 @@ def call(env)
 					# Don't retain persistent connections unless they will eventually be closed:
 					@internet.close unless @persistent
 				end
+
+				private
+
+				def with_timeout
+					if @timeout
+						Async::Task.current.with_timeout(@timeout) do
```

Comment:
If you think it's better, you can supply the kind of exception to raise as the 2nd argument.

https://github.com/socketry/async/blob/a9ca119df23f319f2c7a8e3f2cfc8862157381ee/lib/async/reactor.rb#L279

Reply:
yes, this looks cleaner, thanks!